src/nvim/regexp_defs.h

```diff
@@ -32,7 +32,16 @@
 
 // In the NFA engine: how many states are allowed.
 #define NFA_MAX_STATES 100000
-#define NFA_TOO_EXPENSIVE -1
+
+/// Result of a regex search.
+///
+/// kRegexMatchNotFound and kRegexMatchesFound may also be used for their
+/// integer values.
```

Comment:
I can take a look at what the impact would be to change the return type for the one function and then just use -1/0/>0 for the rest.  After all FALSE isn't any more descriptive than 0 in that case.